sqlx-postgres/src/testing/mod.rs

```diff
@@ -180,16 +182,25 @@ async fn test_context(args: &TestArgs) -> Result<TestContext<Postgres>, Error> {
 }
 
 async fn do_cleanup(conn: &mut PgConnection, created_before: Duration) -> Result<usize, Error> {
-    // since SystemTime is not monotonic we added a little margin here to avoid race conditions with other threads
-    let created_before = i64::try_from(created_before.as_secs()).unwrap() - 2;
-
-    let delete_db_names: Vec<String> = query_scalar(
-        "select db_name from _sqlx_test.databases \
+    let delete_db_names: Vec<String> = if let Some(current_run_id) = get_current_run_id() {
+        query_scalar(
+            "select db_name from _sqlx_test.databases \
+            where test_run_id::text != $1 or test_run_id is null",
+        )
+        .bind(current_run_id)
+        .fetch_all(&mut *conn)
+        .await?
+    } else {
+        // since SystemTime is not monotonic we added a little margin here to avoid race conditions with other threads
```

Comment:
If we get a `CARGO_RUN_ID` this whole branch can be removed as well as the timestamp from the db.

Reply:
I just realized, Cargo only runs one test binary at a time, right? Instead of a timestamp, we could just store a random run ID in-memory if one isn't passed in.

Reply:
Yup, only one test binary at a time.

### Problem:
Currently we're using the presence of an uuid to identify if we need to fallback to timestamp cleanup.
I considered your idea earlier, but I realized that would mean that any other test runners would have more races.
Ie, a test runner that is not `cargo test` and not `cargo nextest`.

### possible solutions
* Accept it(are there even any other test runners?)
* Identify if normal `cargo test` is running and only then assign uuid. (will have a check on this one)